geronimo-openapi-impl/src/main/java/org/apache/geronimo/microprofile/openapi/impl/processor/AnnotationProcessor.java

```diff
@@ -881,18 +882,32 @@ private Encoding mapEncoding(final Supplier<org.eclipse.microprofile.openapi.mod
         }
         if (annotatedElement != null) {
             if (annotatedElement.isAnnotationPresent(HeaderParam.class)) {
+                final HeaderParam annotation = annotatedElement.getAnnotation(HeaderParam.class);
                 impl.in(org.eclipse.microprofile.openapi.models.parameters.Parameter.In.HEADER);
+                mapParameterName(impl, annotation.value());
             } else if (annotatedElement.isAnnotationPresent(CookieParam.class)) {
+                final CookieParam annotation = annotatedElement.getAnnotation(CookieParam.class);
                 impl.in(org.eclipse.microprofile.openapi.models.parameters.Parameter.In.COOKIE);
+                mapParameterName(impl, annotation.value());
             } else if (annotatedElement.isAnnotationPresent(PathParam.class)) {
+                final PathParam annotation = annotatedElement.getAnnotation(PathParam.class);
                 impl.in(org.eclipse.microprofile.openapi.models.parameters.Parameter.In.PATH);
+                mapParameterName(impl, annotation.value());
             } else if (annotatedElement.isAnnotationPresent(QueryParam.class)) {
+                final QueryParam annotation = annotatedElement.getAnnotation(QueryParam.class);
                 impl.in(org.eclipse.microprofile.openapi.models.parameters.Parameter.In.QUERY);
+                mapParameterName(impl, annotation.value());
             }
         }
         return impl;
     }
 
+    private void mapParameterName(final ParameterImpl impl, final String name) {
+        if (StringUtils.isEmpty(impl.getName()) && !StringUtils.isEmpty(name)) {
```

Comment:
Can name be empty? Dont think so and zvzn if so it would be the same so second test is useless right?

Reply:
Yes, name could be empty, one of legitimate examples is `@QueryParam("")` when binding to the bean class is used.

Reply:
Even if the name is empty, we still could simplify the condition, did that, thanks.